R/any_is_na_linter.R

```diff
@@ -45,15 +45,28 @@ any_is_na_linter <- function() {
     ]
   "
 
+  in_xpath <- "//SPECIAL[text() = '%in%']/preceding-sibling::expr[NUM_CONST[starts-with(text(), 'NA')]]"
```

Comment:
Do we want to allow linting custom NA sentinels, too? e.g. `lubridate::NA_Date_`.

Reply:
I think this usage is rare enough that I would only do so by later request... there are no hits for that:

https://github.com/search?q=lang%3AR%20%2FNA_date_%5Cs*%5B%25%5Din%5B%25%5D%2F&type=code

Nor are there any hits for `NA_*_ %in%` besides the defaults:

https://github.com/search?q=lang%3AR+%2FNA_%5B%5Eirc%5D%5Ba-z%5D%2B_%5Cs*%5B%25%5Din%5B%25%5D%2F&type=code